extensions/platform-http/runtime/src/main/java/org/apache/camel/quarkus/component/platform/http/runtime/QuarkusPlatformHttpConsumer.java

```diff
@@ -0,0 +1,359 @@
+import java.io.StringWriter;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import io.vertx.core.MultiMap;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.ext.web.Route;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+
+import org.apache.camel.Consumer;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.NoTypeConversionAvailableException;
+import org.apache.camel.Processor;
+import org.apache.camel.TypeConversionException;
+import org.apache.camel.TypeConverter;
+import org.apache.camel.component.platform.http.PlatformHttpComponent;
+import org.apache.camel.component.platform.http.PlatformHttpEndpoint;
+import org.apache.camel.component.platform.http.spi.Method;
+import org.apache.camel.spi.HeaderFilterStrategy;
+import org.apache.camel.support.DefaultConsumer;
+import org.apache.camel.support.DefaultMessage;
+import org.apache.camel.support.ExchangeHelper;
+import org.apache.camel.support.MessageHelper;
+import org.apache.camel.support.ObjectHelper;
+
+/**
+ * A Quarkus specific {@link Consumer} for the {@link PlatformHttpComponent}.
+ */
+public class QuarkusPlatformHttpConsumer extends DefaultConsumer {
+    private final Router router;
+    private final Object routeLock = new Object();
+    private Route route;
+
+    public QuarkusPlatformHttpConsumer(Endpoint endpoint, Processor processor, Router router) {
+        super(endpoint, processor);
+        this.router = router;
+    }
+
+    @Override
+    public PlatformHttpEndpoint getEndpoint() {
+        return (PlatformHttpEndpoint) super.getEndpoint();
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+
+        synchronized (routeLock) {
+            route = router.route(getEndpoint().getPath());
+            final Set<Method>methods = getEndpoint().getEffectiveMethods();
+            if (!methods.equals(Method.getAll())) {
+                methods.stream().forEach(m -> route.method(HttpMethod.valueOf(m.name())));
+            }
+
+            route.handler(ctx -> {
+                try {
+                    Exchange e = toExchange(ctx, getEndpoint().createExchange(),
+                            getEndpoint().getHeaderFilterStrategy());
+                    getProcessor().process(e);
+                    writeResponse(ctx, e, getEndpoint().getHeaderFilterStrategy());
+                } catch (Exception e1) {
+                    throw new RuntimeException(e1);
+                }
+            });
+        }
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        synchronized (routeLock) {
```

Comment:
don't think the lock is needed as `doStop` is atomic

Reply:
+1